File: springboot-jwt/src/main/java/com/grepp/spring/infra/auth/token/TokenCookieFactory.java
package com.grepp.spring.infra.auth.token;

import org.springframework.http.ResponseCookie;

public class TokenCookieFactory {

    public static ResponseCookie create(String name, String value, Long expires){
        return ResponseCookie.from(name, value)
            .httpOnly(true)
            .maxAge(expires/1000)
            // todo: 배포 후 true 로 변경
            .secure(false)
            .path("/")
            .build();
    }

}

File: grepp-gateway/src/main/kotlin/com/grepp/spring/token/util/TokenCookieFactory.kt
package com.grepp.spring.token.util

import com.grepp.spring.token.code.TokenType
import org.springframework.http.ResponseCookie

object TokenCookieFactory {
    fun create(name: String, value: String, expires: Long): ResponseCookie {
        return ResponseCookie.from(name, value)
            .httpOnly(true)
            .maxAge(expires) // todo : 배포 후 true 로 변경
            .secure(false)
            .path("/")
            .build()
    }

    fun createExpiredToken(tokenType: TokenType): ResponseCookie {
        return ResponseCookie.from(tokenType.name, "")
            .httpOnly(true)
            .maxAge(0)
            .secure(false)
            .path("/")
            .build()
    }
}
